src/core/a-animation.js

```diff
@@ -373,6 +379,19 @@ function getAnimationValues (el, attribute, dataFrom, dataTo, currentValue) {
   };
 
   /**
+   * Match the scema type to color
+   * @return {bool} if the schema is of type color
+   */
+  function isColor () {
+    var componentName = attributeSplit[0];
+    var propertyName = attributeSplit[1];
+    var component = el.components[componentName];
+    var schema = component && component.schema;
+    if (schema) { return schema[propertyName].type === 'color'; }
```

Comment:
@ngokevin Did you ever get a chance to check out the branch?  Also was wondering if going in this direction for the animation should maybe be rethought?  Having to add special cases for each animation type seems unintuitive.  Couldn't we do a more effective remapping of primitives to their object by passing a reference?

Just worried this branching if of detecting type is going to get uglier as we try to support different interpolation types.  I built out something similar in a repo where I was trying to essentially build out an aframe-like framework (before I heard of aframe) in meteor.  

Through this I had build out a transitioner object that would take a payload and tween any object with numeric values.  As long as you passes a string describing it such as `mesh.material.specular` and a ending object `{ r: 1, g: 0, b: .4 }`.

Here is a reference to the code and how I approached it.  Not sure if it would help https://github.com/poetic/poetic-vr-boilerplate/blob/master/utils-animate.js.